frontend/catalyst/compiler.py

```diff
@@ -242,7 +242,8 @@ def lower_mhlo_to_linalg(filename):
     Returns:
         a path to the output file
     """
-    assert filename[-5:] == ".mlir", "input is not an mlir file"
+    if filename.split(os.extsep)[-1] != "mlir":
+        raise ValueError(f"Input file ({filename}) is not a MLIR file")
```

Comment:
Might be nice to be more specific here:
```suggestion
        raise ValueError(f"Input file ({filename}) for MHLO lowering is not an MLIR file")
```